core/services/pipeline/common.go

```diff
@@ -169,17 +171,38 @@ type JSONSerializable struct {
 	Null bool
 }
 
+// UnmarshalJSON implements custom unmarshaling logic
 func (js *JSONSerializable) UnmarshalJSON(bs []byte) error {
 	if js == nil {
 		*js = JSONSerializable{}
 	}
-	return json.Unmarshal(bs, &js.Val)
+
+	if err := json.Unmarshal(bs, &js.Val); err != nil {
+		return err
+	}
+
+	// This "hack" is needed to solve the issue related to "\u0000" characters that should be stored in DB.
+	// PostgreSQL does not allow to store such characters so it should be HEX encoded.
+	if val, ok := js.Val.(string); ok {
```

Comment:
I do not like this workaround. We should serialize all `[]byte` in a sensible way instead.

Reply:
I don't like it too. But as I mentioned above, we may have two cases:

- Bytes array that should be HEX encoded and HEX decoded.
- HEX encoded string so it should not be decoded in the unmarshal json function.